cache.go

```diff
@@ -0,0 +1,98 @@
+package wazero
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	goruntime "runtime"
+
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/internal/compilationcache"
+	"github.com/tetratelabs/wazero/internal/version"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// CompileCache is the configuration for compilation cache behavior of wazero.Runtime and can be passed to wazero.RuntimeConfig.
+type CompileCache interface{ api.Closer }
+
+// NewCache returns a new Cache to be passed to RuntimeConfig.
+// This configures only in-memory cache, and doesn't persist to the file system. See wazero.NewCompileCacheWithDir for detail.
+//
+// The returned CompileCache can be used to share the in-memory compilation results across multiple instances of wazero.Runtime.
+func NewCache() CompileCache {
+	return &cache{}
+}
+
+// NewCompileCacheWithDir is the same as Cache returned by wazero.NewCache except that this also persists
+// the compilation results into the directory specified by `dirname` parameter.
+//
+// If the dirname doesn't exist, this creates the directory. And if the directory creation fails, this returns an error.
+//
+// If the given directory already exists, and it's a file, this returns an error.
```

Comment:
probably not needed to clarify this, we can spend the words on  why to use this instead.
```suggestion
Those running wazero as a CLI or frequently restarting a process using the same wasm should
use this feature to reduce time waiting to compile the same module a second time.
```